pkg/cmd/node/node_diagnostics.go

```diff
@@ -14,6 +17,12 @@ func newCmdNodeDiagnostics(ctx api.Context) *cobra.Command {
 		Use:   "diagnostics",
 		Short: "Use diagnostics bundles",
 		RunE: func(cmd *cobra.Command, args []string) error {
+
+			_, ok := ctx.EnvLookup(cli.EnvStrictDeprecations)
+			if ok {
+				return fmt.Errorf("unknown command diagnostics")
+			}
+			ctx.Deprecated("This command is deprecated since DC/OS 1.14, please use 'dcos diagnostics' instead.")
```

Comment:
The env check is done in the Deprecated function, it can be removed to just have:

```suggestion
			err := ctx.Deprecated("This command is deprecated since DC/OS 1.14, please use 'dcos diagnostics' instead.")
			if err != nil {
				return err
			}
```

Reply:
Then https://github.com/dcos/dcos-core-cli/blob/1.14-patch.x/pkg/cmd/service/service.go#L18 needs a fix for this b/c I looked it up there. :-)

Reply:
The use-case is different there, as it's not deprecating the command altogether but changing its behavior (printing the help menu instead of showing a list).